library/core/src/alloc/layout.rs

```diff
@@ -117,6 +121,9 @@ impl Layout {
     #[must_use]
     #[inline]
     #[rustc_allow_const_fn_unstable(ptr_alignment_type)]
+    #[requires(align > 0)]
+    #[requires((align & (align - 1)) == 0)]
+    #[requires(size <= isize::MAX as usize - (align - 1))]
```

Comment:
I'm curious, what do you think about:
```rust
#[requires(from_size_align(size, align).is_ok())]
```

Reply:
Do we support function calls in here at this time?

Reply:
Yes, as long as they are side-effect free.

Reply:
Thanks, done.